docs/topics/patterns/instant-messaging.mdx

```diff
@@ -0,0 +1,359 @@
+
+* Paging messages N at a time
+* Adding multiple messages at once
+* Maintaining a chat history of fixed length, i.e. the last 100 messages
+
+When using a list, you have more array manipulation functions than with a sorted set, allowing you to automatically truncate message history if it gets too long or remove specific elements for things like content moderation.
+
+### Delivery
+
+Technically, message storage is not a necessary component for instant messaging. In theory you could have a messaging system that only showed the messages that were sent while a user is connected. Messages could be stored in memory in the client and you'd be done. That said, message delivery is a mandatory part of instant messaging. Getting the message from the sender to all recipients with minimal latency is the core of this pattern. To do this, we use [Momento Topics](https://www.gomomento.com/platform/topics).
+
+Momento Topics is a low-latency pub/sub service that connects clients together, clients to servers, servers to clients, and servers to servers. Think WebSockets but *way better*. When you publish a message to a topic, Momento will deliver it to all subscribers of the topic, providing fan-out capability or 1 to 1, depending on your use case.
+
+For instant messaging, the sender will *publish* a message to a topic. The recipients of the message, be it one or many depending on your use case, will be *subscribers*. Momento Topics will deliver the published message to the subscribers in real-time, providing for the instant messaging experience we're looking for!
+
+:::tip
+Topics are dynamic and are not defined resources in your infrastructure. The recommended way to build instant messaging is to *use the chat room or session identifier as the topic name*. This provides flexible and dedicated topics for chat use cases.
+:::
+
+#### Publishing a message
+
+To publish a message, all you need is a single command:
+
+<Tabs>
+  <TabItem value="node" label="Node.js">
+
+  ```javascript
+  await topics.publish('message-namespace', sessionId, 'Hello world!');
+  ```
+  </TabItem>
+  <TabItem value="python" label="Python">
+
+  ```python
+  topics.publish("message-namespace", sessionId, "Hello world!")
+  ```
+  </TabItem>
+  <TabItem value="go" label="Go">
+
+  ```go
+  _, err := topics.Publish(ctx, &momento.TopicPublishRequest{
+    CacheName: momento.String("message-namespace"),
+    TopicName: sessionId,
+    Value: momento.String("Hello world!")
+  })
+  ```
+  </TabItem>
+  <TabItem value="java" label="Java">
+
+  ```java
+  final TopicPublishResponse response = topics.publish("message-namespace", sessionId, "Hello world!").join();
+  ```
+  </TabItem>
+  <TabItem value="dotnet" label=".NET">
+
+  ```csharp
+  var response = await topics.PublishAsync("message-namespace", sessionId, "Hello world!");
+  ```
+  </TabItem>
```

Comment:
this is a nit and not a blocker for this PR, but it'd be great to define an example snippet for this, like what we do elsewhere in the docs. (e.g. https://docs.momentohq.com/topics/develop/api-reference#publish ). It would help us make sure that the code actually compiles and help us keep track of where we are missing examples for given languages. But I realize that it is a bit of extra work to go put the snippets into the SDK repos.